### PR TITLE
LUN-6: Add Theming

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

File: src/components/__tests__/Button.spec.tsx [hunk 1]
```diff
@@ -1,88 +1,93 @@
...
 import { BUTTONS_THEME } from '../../constants/data'
+import theme from '../../constants/theme'
+import { COLORS } from '../../constants/theme/colors'
+import { mocked } from 'ts-jest/utils'
+import wrapWithTheme from '../../testing/wrapWithTheme'
+
+type ButtonPropsType = ComponentProps<typeof Button>
+
+const expectStylesToMatch = (component: ShallowWrapper<ButtonPropsType>, expectedStyles: Record<string, any>): void => {
+  expect(component.dive().dive().dive().prop<Record<string, any>>('style')[0]).toEqual(
```
Comment (Taggotty) [inline src/components/__tests__/Button.spec.tsx:14]: Not really pretty. Apparently there is a [package for testing styled components ](https://github.com/styled-components/jest-styled-components). But it is currently not working for native. I'll keep my eyes open on this.

File: tsconfig.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "lib": ["es6"],                           /* Specify library files to be included in the compilation. */
     "allowJs": true,                          /* Allow javascript files to be compiled. */
     // "checkJs": true,                       /* Report errors in .js files. */
-    "jsx": "react-native",                    /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
+    "jsx": "react",                           /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
     // "declaration": true,                   /* Generates corresponding '.d.ts' file. */
```
Comment (Taggotty) [inline tsconfig.json:11]: Don't know why you would use react-native. In all the setup tutorials the use react and ts-jest basically also only works with react here...

**Review by SteffiStuffel — APPROVED** [review]
:tada:

File: src/components/AudioPlayer.tsx [hunk 3]
```diff
@@ -91,8 +100,12 @@ const AudioPlayer = (props: AudioPlayerProps): ReactElement => {
 
   return (
     <StyledView>
-      <VolumeIcon testID='volume-button' disabled={disabled} onPress={() => handleSpeakerClick(document?.audio)}>
-        <VolumeUp fill={disabled ? COLORS.lunesBlackUltralight : isActive ? COLORS.lunesRed : COLORS.lunesRedDark} />
+      <VolumeIcon
+        theme={theme}
```
Comment (klinzo) [inline src/components/AudioPlayer.tsx:104]: Passing `theme` as prop should not be necessary, is it?

File: src/navigation/NavigationTypes.ts [hunk 4]
```diff
@@ -1,5 +1,7 @@
 import { DocumentsType, DocumentType } from '../constants/endpoints'
 import { ExerciseKeyType, ResultType, SimpleResultType } from '../constants/data'
+import { SvgProps } from 'react-native-svg'
+import { ComponentType } from 'enzyme'
```
Comment (klinzo) [inline src/navigation/NavigationTypes.ts:4]: Is importing the type from `enzyme` correct here?

File: .circleci/config.yml [hunk 5]
```diff
@@ -197,7 +197,7 @@ jobs:
                 root: ios/output/gym
     bump_version:
         docker:
-            - image: circleci/node:12.16.3
+            - image: circleci/node:14.16.0
```
Comment (SteffiStuffel) [inline .circleci/config.yml:200]: Is it possible these updates make the build a lot slower? Before the e.g. check-step takes aroung 30sec and now more than a few minutes.

Comment (Taggotty) [reply .circleci/config.yml:200]: fixed after adding `--maxWorkers=1` to yarn test

**Pull request merged**